specification/paths/Onboarding-v1-invitations-invitation_id-onboard.json

```
@@ -0,0 +1,97 @@
+{
+  "parameters": [
+    {
+      "$ref": "#/components/parameters/path-invitation_id"
+    }
+  ],
+  "post": {
+    "tags": [
+      "RPC"
+    ],
+    "security": [
+      {
+        "OAuth2": [
+          "brokers.manage"
+        ]
+      }
+    ],
+    "summary": "Start onboarding process.",
+    "description": "This endpoint creates a user based on an invitation.",
+    "requestBody": {
+      "description": "The onboarding request.",
+      "required": true,
+      "content": {
+        "application/vnd.api+json": {
+          "schema": {
+            "type": "object",
+            "required": [
+              "data"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "data": {
+                "required": [
+                  "username",
+                  "password",
+                  "first_name",
+                  "last_name",
+                  "email",
+                  "shop_name",
+                  "address"
+                ],
+                "properties": {
+                  "username": {
+                    "type": "string",
+                    "example": "Sherlock"
+                  },
+                  "password": {
+                    "type": "string",
+                    "example": "The best passwords are long, so use a sentence!"
+                  },
+                  "first_name": {
+                    "type": "string",
+                    "example": "Sherlock"
+                  },
+                  "last_name": {
+                    "type": "string",
+                    "example": "Holmes"
+                  },
+                  "email": {
+                    "$ref": "#/components/schemas/Email",
+                    "example": "s.holmes@holmesinvestigations.com"
+                  },
+                  "company": {
+                    "type": "string",
+                    "example": "Holmes Investigations"
+                  },
+                  "phone_number": {
+                    "type": "string",
+                    "example": "+44 20 7946 0958"
+                  },
+                  "shop_name": {
+                    "type": "string",
+                    "example": "Holmes Investigations"
+                  },
+                  "address": {
+                    "$ref": "#/components/schemas/BaseAddress"
+                  },
```

You can make `address` fields required by defining a second object it needs to adhere to.
Same goes for the `return_address` fields.
```suggestion
                  "address": {
                    "allOf": [
                      {
                        "$ref": "#/components/schemas/BaseAddress"
                      },
                      {
                        "required": [
                          "street_1",
                          "city",
                          "country_code"
                        ]
                      },
                    ]
                  },
```